Client/src/ArmoniK.DevelopmentKit.Client.Common/Submitter/BaseClientSubmitter.cs

```diff
@@ -514,39 +528,51 @@ public void WaitForTasksCompletion(IEnumerable<string> taskIds)
   public async Task<byte[]> TryGetResultAsync(ResultRequest     resultRequest,
                                               CancellationToken cancellationToken = default)
   {
-    var streamingCall = ControlPlaneService.TryGetResultStream(resultRequest,
-                                                               cancellationToken: cancellationToken);
-    var chunks = new List<ReadOnlyMemory<byte>>();
-    var len    = 0;
+    await mutex_.WaitAsync(cancellationToken);
+
+    List<ReadOnlyMemory<byte>> chunks;
+    int                        len;
 
-    while (await streamingCall.ResponseStream.MoveNext(cancellationToken))
+    try
     {
-      var reply = streamingCall.ResponseStream.Current;
+      var streamingCall = ControlPlaneService.TryGetResultStream(resultRequest,
+                                                                 cancellationToken: cancellationToken);
+      chunks = new List<ReadOnlyMemory<byte>>();
+      len    = 0;
 
-      switch (reply.TypeCase)
+      while (await streamingCall.ResponseStream.MoveNext(cancellationToken))
       {
-        case ResultReply.TypeOneofCase.Result:
-          if (!reply.Result.DataComplete)
-          {
-            chunks.Add(reply.Result.Data.Memory);
-            len += reply.Result.Data.Memory.Length;
-          }
+        var reply = streamingCall.ResponseStream.Current;
 
-          break;
-        case ResultReply.TypeOneofCase.None:
-          return null;
+        switch (reply.TypeCase)
+        {
+          case ResultReply.TypeOneofCase.Result:
+            if (!reply.Result.DataComplete)
+            {
+              chunks.Add(reply.Result.Data.Memory);
+              len += reply.Result.Data.Memory.Length;
+            }
 
-        case ResultReply.TypeOneofCase.Error:
-          throw new Exception($"Error in task {reply.Error.TaskId} {string.Join("Message is : ", reply.Error.Errors.Select(x => x.Detail))}");
+            break;
+          case ResultReply.TypeOneofCase.None:
+            return null;
 
-        case ResultReply.TypeOneofCase.NotCompletedTask:
-          return null;
+          case ResultReply.TypeOneofCase.Error:
+            throw new Exception($"Error in task {reply.Error.TaskId} {string.Join("Message is : ", reply.Error.Errors.Select(x => x.Detail))}");
 
-        default:
-          throw new ArgumentOutOfRangeException("Got a reply with an unexpected message type.",
-                                                (Exception)null);
+          case ResultReply.TypeOneofCase.NotCompletedTask:
+            return null;
+
+          default:
+            throw new ArgumentOutOfRangeException("Got a reply with an unexpected message type.",
+                                                  (Exception)null);
+        }
       }
     }
+    finally
+    {
+      mutex_.Release();
+    }
 
     var res = new byte[len];
```

Comment:
No way to avoid a new allocation here ?

Reply:
Currently, no. Besides, it is out of scope of this PR.